### ISSUE TITLE
Read_GML() causes segmentation fault on macOS Sierra 10.12.3 and Ubuntu 12.04.5

### ISSUE THREAD

**Issue by isokrates** [igraph/igraph#977]
Hi,

I have installed igraph 0.7.1 on macOS Sierra 10.12.3 and igraph 0.6.5 on Ubuntu 12.04.5, and have noticed the same problem on both platforms. The following code causes a segmentation fault:

`import igraph as ig`
`G = ig.Graph.Read_GML('test.gml')`

Here is a link to the crash data I had the option to send to Apple, just in case it's useful:

https://www.dropbox.com/scl/fi/ak6fedcg2azgxy0bolrrb/segfault.txt?dl=0

Any idea why this is happening, or what I can do to fix it?

Thanks!

**Comment by gaborcsardi** [comment]
It is probably an invalid file.

**Comment by ntamas** [comment]
I concur. Please send us the file if possible so we can check it.

**Comment by isokrates** [comment]
Thanks! I discovered that I don't have the problem if I make all the IDs integers (Gephi doesn't say anything about this, though, and I used Gephi to export). I'm still not sure why it would cause a segfault, though.

I now have a new problem...

`>>> import igraph as ig`
`>>> ig.Graph.Read_GML('test.gml')`
`Traceback (most recent call last):`
`  File "<stdin>", line 1, in <module>`
`igraph._igraph.InternalError: Error at foreign.c:1121: Parse error in GML file, line 1 (syntax error, unexpected NUM, expecting KEYWORD or LISTCLOSE), Parse error`

Here's my test file: https://www.dropbox.com/scl/fi/uzv6n7044umjookt5pa5y/test.gml?dl=0

I don't know what is wrong with the file, and I have compared it to some files that do work. Any ideas?
